### PR TITLE
Enable more complex metric calculations

### PR DESCRIPTION
**Pull request by harrisonpim**
Adds terraform for [#3492](https://github.com/wellcometrust/platform/issues/3942) and a basic example container.

### CODE REVIEW

File: ecs_tasks/search_relevance_explicit_NDCG/src/search_relevance_explicit_NDCG.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    es_client=es_client,
+    index='tracking_search'
+)
+
+# process events and send them to the rolled elasticsearch index
+for group_id, event_group in split_events(all_events):
+    transformed_data = transform_event_group(event_group)
+    if transformed_data is not None:
+        es_client.index(
+            index='search_relevance_explicit_NDCG',
```
Comment (jamesgorrie) [inline ecs_tasks/search_relevance_explicit_NDCG/src/search_relevance_explicit_NDCG.py]: Could we keep this to lowercase? [ES doesn't support uppercase ](https://github.com/elastic/elasticsearch/issues/31262).

We should probably reflect that in the folder name too unless there is a reason for the uppercase?

File: ecs_tasks/search_relevance_explicit_NDCG/src/elasticsearch_utils.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import certifi
+import pandas as pd
+from elasticsearch import Elasticsearch
+
+today = datetime.today()
+yesterday = today - timedelta(days=1)
+
+
+def stringify(timestamp):
+    return str(timestamp).replace(' ', 'T')
```
Comment (jamesgorrie) [inline ecs_tasks/search_relevance_explicit_NDCG/src/elasticsearch_utils.py]: Where does this differentiation come from?
If we need it - maybe `stringify_timestamp` might make more sense.

File: ecs_tasks/search_relevance_explicit_NDCG/src/search_relevance_explicit_NDCG.py (hunk 1 above)
Comment (harrisonpim) [reply ecs_tasks/search_relevance_explicit_NDCG/src/search_relevance_explicit_NDCG.py]: no strong reason for uppercasing, just makes it clearer that NDCG stands for something and isn't just keysmashingljashduygd - will lowercase both

File: ecs_tasks/search_relevance_explicit_NDCG/src/elasticsearch_utils.py (hunk 2 above)
Comment (harrisonpim) [reply ecs_tasks/search_relevance_explicit_NDCG/src/elasticsearch_utils.py]: During testing, ES wasn't happy with timestamps without the `T` separator. Bit hacky, but it works

**Comment by jamieparkinson** [comment]
What's going on with this PR?

**Comment by harrisonpim** [comment]
This was left in an unfinished state when it was decided that this work was too developer-y to be a good use of my time. The remaining work was never assigned to anyone else, and it's been sat in limbo since then. It blocks the more complicated reporting metrics which can't otherwise be calculated in kibana, but given that we haven't had people complaining about it and i can still calculate those metrics myself, it might not be worth reprioritising.

**Comment by jamieparkinson** [comment]
Is the work here broken / would it break things if merged? ie, could the remaining work (which I guess is what's left on the checklist in https://github.com/wellcometrust/platform/issues/3942) be done in a separate PR and ticket? If so I'd merge it and split that remaining work into another ticket. If not, maybe close the PR.

**Pull request closed without merging**